lib/matplotlib/pyplot.py

```diff
@@ -1313,19 +1313,19 @@ def subplot_tool(targetfig=None):
     """
     Launch a subplot tool window for a figure.
 
-    A :class:`matplotlib.widgets.SubplotTool` instance is returned.
+    A `matplotlib.widgets.SubplotTool` instance is returned.
     """
     if targetfig is None:
         targetfig = gcf()
-
-    with rc_context({'toolbar': 'None'}):  # No nav toolbar for the toolfig.
-        toolfig = figure(figsize=(6, 3))
-    toolfig.subplots_adjust(top=0.9)
-
-    if hasattr(targetfig.canvas, "manager"):  # Restore the current figure.
-        _pylab_helpers.Gcf.set_active(targetfig.canvas.manager)
-
-    return SubplotTool(targetfig, toolfig)
+    with rc_context({"toolbar": "none"}):  # No navbar for the toolfig.
+        # Use new_figure_manager() instead of figure() so that the figure
+        # doesn't get registered with pyplot.
```

Comment:
I'm wondering why this whole function lives in `pyplot`. It should either be private or live in `backend_bases` or similar.

Reply:
Probably because someone wanted to call plt.subplot_tool() at some point back in the let's do everything in pyplot era.  Not going to change this here...